core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -238,6 +296,26 @@ public void start(Boolean isPrimary) throws IOException {
             ThreadFactoryUtils.build("DailyMetadataBackup-%d", true)), mUfs);
         mDailyBackup.start();
       }
+      if (ServerConfiguration.getBoolean(PropertyKey.MASTER_UPDATE_CHECK_ENABLED)) {
+        if (mState.getClusterID() == null) {
```

Comment:
I feel like a cluster id is generally useful, so could this be independent from the version check?

Reply:
sgtm